apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.SingleChannel.cs

```diff
@@ -392,13 +401,34 @@ private void HandlePullMessageData(Task<bool> moveNextTask)
                 receivedMessages.Clear();
                 // Get all ack-ids, used to extend leases as required.
                 var msgIds = new HashSet<string>(msgs.Select(x => x.AckId));
+                _receiptModAckForExactlyOnceDelivery = _exactlyOnceDeliveryEnabled;
```

Comment:
The block starting at line 382 is now pretty huge. We should probably refactor it - but maybe only after getting this PR in. (Otherwise it'll be harder to review.)